shell/platform/glfw/key_event_handler.cc

```diff
@@ -5,21 +5,68 @@
 #include "flutter/shell/platform/glfw/key_event_handler.h"
 
 #include <iostream>
+#include <map>
 
 #include "flutter/shell/platform/common/cpp/client_wrapper/include/flutter/json_message_codec.h"
 
 static constexpr char kChannelName[] = "flutter/keyevent";
 
 static constexpr char kKeyCodeKey[] = "keyCode";
 static constexpr char kKeyMapKey[] = "keymap";
+static constexpr char kScanCodeKey[] = "scanCode";
+static constexpr char kModifiersKey[] = "modifiers";
 static constexpr char kTypeKey[] = "type";
+static constexpr char kToolkitKey[] = "toolkit";
+static constexpr char kUnicodeScalarValuesProduced[] =
+    "unicodeScalarValuesProduced";
+
+static constexpr char kLinuxKeyMap[] = "linux";
+static constexpr char kGLFWKey[] = "glfw";
 
-static constexpr char kAndroidKeyMap[] = "android";
 static constexpr char kKeyUp[] = "keyup";
 static constexpr char kKeyDown[] = "keydown";
 
 namespace flutter {
 
+// Converts a utf8 const char* to a 32 bit int. The Flutter framework accepts
+// only one 32 bit int, therefore, only up to 4 bytes are accepted. If it's
+// larger than 4 bytes, only the first 4 will be used.
+std::uint32_t DecodeUTF8(const char* utf8) {
+  size_t length = strlen(utf8);
+  // Defend against > 4 byte code points.
+  length = length >= 4 ? length : length;
+
+  // Tracks how many bits the current byte should shift to the left.
+  int shift = length;
+
+  // All possible masks for a 4 byte utf8 code point.
+  std::map<int, int> masks;
```

Comment:
Why not use a vector for these rather than a map, since you are always only using the first 4 positions? (You'd want to zero-index rather than one-index, obviously)

Reply:
Done